tests/e2e/convolution/generate_e2e_conv2d_tests.py

```diff
@@ -37,6 +51,55 @@ class ShapesId(enum.Enum):
     SMALL = "small"
     MEDIUM = "medium"
     LARGE = "large"
+    GPU_LARGE = "gpu_large"
+
+
+# Describes a workgroup and tiling schedule to target a specific MMA intrinsic.
+@dataclasses.dataclass
+class MMASchedule:
+    intrinsic: str
+    m_count: int  # Number of subgroups per workgroup along M
+    n_count: int  # Number of subgroups per workgroup along N
+    m_tile_count: int
+    n_tile_count: int
+    k_tile_count: int
+
+    def __str__(self):
+        return (
+            "mma_schedule = #iree_gpu.mma_schedule<"
+            + f"intrinsic = #iree_gpu.mma_layout<{self.intrinsic}>, "
+            + f"subgroup_m_count = {self.m_count}, "
+            + f"subgroup_n_count = {self.n_count}>"
+        )
+
+
+# Enumerates of the collections of compilation info that we can generate tests
+# for. The values are the accepted values for the --compilation_info= flag.
+@enum.unique
+class CompilationInfoId(enum.Enum):
+    NONE = ""
+    LLVMGPUVectorDistributeMFMA = "LLVMGPUVectorDistributeMFMA"
+    LLVMGPUVectorDistributeWMMA = "LLVMGPUVectorDistributeWMMA"
+    LLVMGPUVectorizeCDNA = "LLVMGPUVectorizeCDNA"
+    LLVMGPUVectorizeRDNA = "LLVMGPUVectorizeRDNA"
+
+
+# Describes how to construct compilation info for the testcase.
+@dataclasses.dataclass
+class CompilationInfo:
+    # Lowering Config
+    tile_sizes: typing.List[typing.List[int]]
+    # Translation Info
+    dispatch_lowering_pass_pipeline: str
+    software_pipeline_depth: int
+    mma_schedule: typing.Optional[MMASchedule]
+    # Compilation info
+    workgroup_size: typing.List[int]
+    subgroup_size: Optional[int] = None
+
+    # Prints the workgroup size
+    def workgroup_size_str(self):
+        return "workgroup_size = [" + ", ".join(map(str, self.workgroup_size)) + "]"
```

Comment:
Are these "compilation infos" part of the public compiler API? How are users expected to make use of these vectorize/vectordistribute MFMA/WMMA/CDNA/RDNA pipelines? I'm worried that this is reaching too deeply into implementation details, when test suites should be modeled more closely after how developers are realistically expected to use the tools.

I would expect:
* program definitions to have computations in them: "multiply these two tensors/matrices with these shapes, then do this other math"
* users/developers to enumerate the explicit devices they want to target, including which features those devices support (e.g. cpu intrinsics, gpu API extensions)
* users/developers to provide tuning configurations per each device that give hints about branches to take during compilation, heuristics to use, etc. (maybe that's this "compilation info"?)

Reply:
I'm also sketched out by `get_rocm_test_compilation_infos()` needing to exist at all here - we shouldn't need anything specific to a single target in a test generator like this.

Reply:
+1 on this. I would expect these e2e tests to look nearly the same as what we have done for winograd convolutions:
https://github.com/iree-org/iree/blob/31bfc932f1428bab08d42d954e6763c0d509376e/tests/e2e/convolution/CMakeLists.txt#L163-L189

You should be able to use the same test generator but choose a different target, and you can add additional compiler flags as needed with `COMPILER_FLAGS`. The test generator should simply create some source IR that performs a convolution, which already exists for winograd. Adding new e2e convolution tests should be able to use that same input IR, but with different compiler specifications. Setting the compilation info is the job of the specific backend codegen, and baking that into the test generator seems like duplicating logic unnecessarily.